installer/msi/build-msi.ps1

```diff
@@ -6,40 +6,113 @@
     .
 .PARAMETER Arch
     The architecture, x86 or x64.
+.PARAMETER Version
+    The version of the driver.
+.PARAMETER VersionLabel
+    The version label of the driver.
 #>
 Param(
-  [string]$Arch,
-  [string]$Version,
-  [string]$VersionLabel,
-  [string]$UpgradeCode
+    [string]$Arch,
+    [string]$Version,
+    [string]$VersionLabel
+    # [string]$UpgradeCode
 )
 
 $ErrorActionPreference = 'Stop'
 
 $ProjectName = "Atlas SQL ODBC"
-$sourceDir = pwd
-$resourceDir = pwd
-$binDir = pwd
+$sourceDir = Get-Location
+$resourceDir = Get-Location
+$binDir = Get-Location
 $objDIr = ".\objs\"
 $WixPath = "C:\wixtools\bin\"
 # for local building, most installations will be in the directory below
-#$WixPath = "C:\Program Files (x86)\WiX Toolset v3.11\bin"
+# $WixPath = "C:\Program Files (x86)\WiX Toolset v3.11\bin"
 $wixUiExt = "$WixPath\WixUIExtension.dll"
 
 # we currently only support x64, but we'll leave the 32 bit support here
 # in case we eventually decide to provide a 32-bit driver
+# upgradeCodes and productCodes are pre-generated for the next several releases 
 if ($Arch -eq "x64") {
-    $upgradeCode = "a4303fe6-c8ca-11ed-afa1-0242ac120002"
-} else {
+    switch ($VersionLabel) {
```

Comment:
and... I took this idea from the postgres installer. Not sure if we want to keep it. It definitely made testing a lot easier.

The pros of this are that we can see the guids easily and they are in source control, and we don't have to look back through logs to see what the codes were. We also aren't relying on the guid utility on the evergreen host to always produce the same guid with the same input (it should, but who knows).

The cons are eventually we'll need to come back and add more guids if this list isn't exhaustive enough.

As an aside, we aren't being fully accurate with this process. 1.0 -> 1.1 is a minor upgrade, but to support the use-case we want, we change the product code to make it a major upgrade.

Reply:
It only matters for snapshot testing

Reply:
I'd prefer just keeping them in uuidgen. This file will grow unbounded like this. It doesn't matter too much either way, though.

Regenerating them is easy because we can just run the uuidgen command from the evergreen file.